GCC Example failed with RHEL 8 & CentOS 8.2

## Description of the problem
**CentOS:**
GCC Example failed with "collect2: fatal error: ld terminated with signal 11 [Segmentation fault], core dumped"

**RHEL**
Graphene-Direct: 
Inconsistency detected by ld.so: dl-call-libc-early-init.c: 37: _dl_call_libc_early_init: Assertion `sym != NULL' failed!
Graphene-SGX:
gcc: error trying to exec '/usr/libexec/gcc/x86_64-redhat-linux/8/cc1': execv: Permission denied

## Steps to reproduce
**For Graphene-Direct**
```
cd Examples/gcc
make -j8 all
make check
```

**For Graphene-SGX**
Need to add these 2 lines in manifest.template for SGX
```
sgx.allowed_files.cc1 = "file:/usr/libexec/gcc/x86_64-redhat-linux/8/cc1"
sgx.allowed_files.collect2 = "file:/usr/libexec/gcc/x86_64-redhat-linux/8/collect2"
```

```
make -j8 SGX=1 all
make SGX=1 check
```

## Expected results
```
make SGX=1 check
Compiling hello.c...
graphene-sgx gcc test_files/helloworld.c -o test_files/hello
error: Using insecure argv source. Graphene will continue application execution, but this configuration must not be used in production!
./test_files/hello
Hello world (./test_files/hello)!
rm -f test_files/hello


Compiling bzip2.c...
graphene-sgx gcc test_files/bzip2.c -o test_files/bzip2
error: Using insecure argv source. Graphene will continue application execution, but this configuration must not be used in production!
rm -f bzip2.tmp
./test_files/bzip2 -z test_files/bzip2.copy
./test_files/bzip2 -d test_files/bzip2.copy.bz2
diff -q test_files/bzip2 test_files/bzip2.copy
rm -f test_files/bzip2 test_file/bzip2.copy


Compiling gzip.c...
graphene-sgx gcc test_files/gzip.c -o test_files/gzip
error: Using insecure argv source. Graphene will continue application execution, but this configuration must not be used in production!
./test_files/gzip test_files/gzip.copy
./test_files/gzip -d test_files/gzip.copy.gz
diff -q test_files/gzip test_files/gzip.copy
rm -f test_files/gzip test_files/gzip.copy
```

## Actual results
**CentOS**
```
[intel1@localhost gcc]$ sudo make check
\n\nCompiling hello.c...
graphene-direct gcc test_files/helloworld.c -o test_files/hello
error: Using insecure argv source. Graphene will continue application execution, but this configuration must not be used in production!
collect2: fatal error: ld terminated with signal 11 [Segmentation fault], core dumped
compilation terminated.
make: *** [Makefile:76: check] Error 1

```


```
[intel1@localhost gcc]$ make SGX=1 check
\n\nCompiling hello.c...
graphene-sgx gcc test_files/helloworld.c -o test_files/hello
error: Using insecure argv source. Graphene will continue application execution, but this configuration must not be used in production!
collect2: fatal error: ld terminated with signal 11 [Segmentation fault], core dumped
compilation terminated.
make: *** [Makefile:76: check] Error 1
```


**RHEL:**
[intel@intel gcc]$ make check
```
\n\nCompiling hello.c...
graphene-direct gcc test_files/helloworld.c -o test_files/hello
error: Using insecure argv source. Graphene will continue application execution, but this configuration must not be used in production!
Inconsistency detected by ld.so: dl-call-libc-early-init.c: 37: _dl_call_libc_early_init: Assertion `sym != NULL' failed!
collect2: error: ld returned 127 exit status
make: *** [Makefile:76: check] Error 1
```


[intel@intel gcc]$ make SGX=1 check
```
\n\nCompiling hello.c...
graphene-sgx gcc test_files/helloworld.c -o test_files/hello
error: Using insecure argv source. Graphene will continue application execution, but this configuration must not be used in production!
gcc: error trying to exec '/usr/libexec/gcc/x86_64-redhat-linux/8/cc1': execv: Permission denied
make: *** [Makefile:76: check] Error 1
[intel@intel gcc]$
```

[centos_gcc_trace.log](https://github.com/oscarlab/graphene/files/6742085/centos_gcc_trace.log)

--- a/Documentation/quickstart.rst
+++ b/Documentation/quickstart.rst
@@ -92,6 +92,12 @@ Running sample applications
 We prepared and tested several applications to demonstrate Graphene and
 Graphene-SGX usability. These applications can be found in the :file:`Examples`
 folder in the repository, each containing a short README with instructions how
-to test it. We recommend starting with simpler, thoroughly documented examples
-like Memcached and Redis, to understand manifest options and features of
-Graphene.
+to test it. We recommend starting with a simpler, thoroughly documented example
+of Redis, to understand manifest options and features of Graphene.
+
+Please note that these sample applications are tested on Ubuntu 18.04 and 20.04.
+Most of these applications are also known to run correctly on
+Fedora/RHEL/CentOS, but with caveats. One caveat is that Makefiles should be
+invoked with ``ARCH_LIBDIR=/lib64 make``. Another caveat is that applications
+that rely on specific versions/builds of Glibc may break (our GCC example is
+known to work only on Ubuntu).

--- a/Examples/gcc/README.md
+++ b/Examples/gcc/README.md
@@ -13,3 +13,12 @@ compiling them from source as some of the other examples do.
 To run the regression tests execute ```make check```. To do the same for SGX,
 execute ```SGX=1 make check```. The regression tests build three sample programs
 - helloworld.c, bzip2.c and gzip.c - and test their functionality.
+
+# Note on OS distributions
+
+It is known that GCC relies on specific Glibc version/build. Because Graphene
+uses its own patched Glibc, this may lead to seemlingly inexplicable failures.
+In particular, this GCC example is known to work on Ubuntu 18.04/20.04 but fail
+on RHEL 8/CentOS 8. There is currently no fix for RHEL/CentOS.
+
+For details, see https://github.com/oscarlab/graphene/issues/2489.

